app/gui/src/ide/integration/project.rs

```diff
@@ -824,15 +828,19 @@ impl Model {
                 None => self.create_node_view(node_info, node_trees, *default_pos),
             }
         }
+        task_handle.end();
         Ok(())
     }
 
     /// Refresh the expressions (e.g., types, ports) for all nodes.
     fn refresh_graph_expressions(&self) -> FallibleResult {
         info!(self.logger, "Refreshing the graph expressions.");
+        let task_handle = profiling::task::start("refresh_graph_expressions");
         let connections = self.graph.connections()?;
         self.refresh_node_views(&connections, false)?;
-        self.refresh_connection_views(connections.connections)
+        let ret = self.refresh_connection_views(connections.connections);
+        task_handle.end();
+        ret
```

Comment:
let's use hierarchical macro here (and other similar places)

Reply:
(see comment above).